invite_tracker.py

```diff
@@ -0,0 +1,105 @@
+        self.logs_channel = config["logs-channel-id"]
+        self.version = "1.0.0"
+
+        self.invites = {}
+        bot.loop.create_task(self.load())
+
+    async def load(self):
+        await self.bot.wait_until_ready()
+        # load the invites
+        for guild in self.bot.guilds:
+            try:
+                self.invites[guild.id] = await guild.invites()
+            except:
+                pass
+
+    def find_invite_by_code(self, inv_list, code):
+        for inv in inv_list:
+            if inv.code == code:
+                return inv
+
+    @commands.Cog.listener()
+    async def on_member_join(self, member):
+        logs = self.bot.get_channel(int(self.logs_channel))
+        eme = Embed(description="Just joined the server", color=0x03d692, title=" ")
+        eme.set_author(name=str(member), icon_url=member.avatar_url)
+        eme.set_footer(text="ID: " + str(member.id))
+        eme.timestamp = member.joined_at
+        try:
+            invs_before = self.invites[member.guild.id]
+            invs_after = await member.guild.invites()
+            self.invites[member.guild.id] = invs_after
+            for invite in invs_before:
+                if invite.uses < self.find_invite_by_code(invs_after, invite.code).uses:
+                    eme.add_field(name="Used invite",
+                                  value=f"Inviter: {invite.inviter.mention} (`{invite.inviter}` | `{str(invite.inviter.id)}`)\nCode: `{invite.code}`\nUses: ` {str(invite.uses)} `", inline=False)
+        except:
+            pass
+        await logs.send(embed=eme)
+
+    @commands.Cog.listener()
+    async def on_member_remove(self, member):
+        logs = self.bot.get_channel(int(self.logs_channel))
+        eme = Embed(description="Just left the server", color=0xff0000, title=" ")
+        eme.set_author(name=str(member), icon_url=member.avatar_url)
+        eme.set_footer(text="ID: " + str(member.id))
+        eme.timestamp = member.joined_at
+        try:
+            invs_before = self.invites[member.guild.id]
+            invs_after = await member.guild.invites()
+            self.invites[member.guild.id] = invs_after
+            for invite in invs_before:
+                if invite.uses > self.find_invite_by_code(invs_after, invite.code).uses:
+                    eme.add_field(name="Used invite",
+                                  value=f"Inviter: {invite.inviter.mention} (`{invite.inviter}` | `{str(invite.inviter.id)}`)\nCode: `{invite.code}`\nUses: ` {str(invite.uses)} `", inline=False)
+        except:
+            pass
+        await logs.send(embed=eme)
+
+    @commands.Cog.listener()
+    async def on_guild_join(self, guild):
+        try:
+            self.invites[guild.id] = await guild.invites()
+        except:
+            pass
+
+    @commands.Cog.listener()
+    async def on_guild_remove(self, guild):
+        try:
+            self.invites.pop(guild.id)
+        except:
+            pass
+
+    # @commands.guild_only()
```

Comment:
Do we need this part? If no, remove it please

Reply:
Done